Dockerfile

```diff
@@ -563,6 +569,13 @@ RUN \
   cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DENABLE_UBSAN=OFF .. && \
   make -j$(nproc) install
 
+RUN \
+  git clone "$LIBGSM_URL" && \
+  cd libgsm && git checkout $LIBGSM_COMMIT && \
+  # special Makefile to buildinstall easier, base64 encoded
+  echo 'Q0ZMQUdTCQk9CS1hbnNpIC1wZWRhbnRpYyAtYyAtTzIgLXMgLUROZWVkRnVuY3Rpb25Qcm90b3R5cGVzPTEgLVdhbGwgLVduby1jb21tZW50IC1EU0FTUiAtRFdBVjQ5IC1ETkRFQlVHIC1JLi9pbmMKClNSQwkJPQkuL3NyYwpPQkpFQ1RTCQk9CSQoU1JDKS9hZGQubwkJXAoJCQkkKFNSQykvY29kZS5vCQlcCgkJCSQoU1JDKS9kZWJ1Zy5vCQlcCgkJCSQoU1JDKS9kZWNvZGUubwkJXAoJCQkkKFNSQykvbG9uZ190ZXJtLm8JXAoJCQkkKFNSQykvbHBjLm8JCVwKCQkJJChTUkMpL3ByZXByb2Nlc3MubwlcCgkJCSQoU1JDKS9ycGUubwkJXAoJCQkkKFNSQykvZ3NtX2Rlc3Ryb3kubwlcCgkJCSQoU1JDKS9nc21fZGVjb2RlLm8JXAoJCQkkKFNSQykvZ3NtX2VuY29kZS5vCVwKCQkJJChTUkMpL2dzbV9leHBsb2RlLm8JXAoJCQkkKFNSQykvZ3NtX2ltcGxvZGUubwlcCgkJCSQoU1JDKS9nc21fY3JlYXRlLm8JXAoJCQkkKFNSQykvZ3NtX3ByaW50Lm8JXAoJCQkkKFNSQykvZ3NtX29wdGlvbi5vCVwKCQkJJChTUkMpL3Nob3J0X3Rlcm0ubwlcCgkJCSQoU1JDKS90YWJsZS5vCgolLm86ICQoU1JDKS8lLmMKCQkJZ2NjICQoQ0ZMQUdTKSAkPwoKbGliZ3NtLmE6ICQoT0JKRUNUUykKCQkJLXJtIGxpYmdzbS5hCgkJCWFyIGNyIGxpYmdzbS5hICQoT0JKRUNUUykKCQkJcmFubGliIGxpYmdzbS5hCgppbnN0YWxsOiBsaWJnc20uYQoJCQlta2RpciAtcCAvdXNyL2luY2x1ZGUvZ3NtLwoJCQljcCBpbmMvKi5oIC91c3IvaW5jbHVkZS9nc20vCgkJCWNwIGxpYmdzbS5hIC91c3IvbGliLwoKY2xlYW46CgkJCXJtIC1mICQoU1JDKS8qLm8gKi5hCgpQSE9OWTogbGliZ3NtLmEK' | base64 -d > Makefile && \
```

Comment:
This file or some easily sed-able file is not available in the repo etc?

Reply:
no, the Makefile is one i created... the original makefile is garbage

Reply:
[Makefile.txt](https://github.com/wader/static-ffmpeg/files/7996078/Makefile.txt)
This is the file

Reply:
Could have it in plaintext also, figured it was best to have it in base64 for transportability

Reply:
Hmm i see, could nearly something like this:
```
CFLAGS="-ansi -pedantic -c -O2 -s -DNeedFunctionPrototypes=1 -Wall -Wno-comment -DSASR -DWAV49 -DNDEBUG -I./inc" \
SRCS=$(echo src/*.c) \
make ${SRCS//.c/.o} && \
ar cr libgsm.a ${SRCS//.c/.o} && \
ranlib libgsm.a && \
... cp ...
```
?

Reply:
SRC=$(echo src/*.c);
    CFLAGS='-c -ansi -pedantic -O2 -s -DNeedFunctionPrototypes=1 -Wall -Wno-comment -DSASR -DWAV49 -DNDEBUG -I./inc';
    gcc ${CFLAGS} ${SRC};
    ar cr libgsm.a ${SRC/\.c/\.o};
    ranlib libgsm.a;
    mkdir -p /usr/include/gsm /usr/lib;
    cp inc/*.h /usr/include/gsm;
    cp libgsm.a /usr/lib;

Thats all i see similiar to what you are refering to... and tit seems to work

one-liner is..

`SRC=$(echo src/*.c); CFLAGS='-c -ansi -pedantic -O2 -s -DNeedFunctionPrototypes=1 -Wall -Wno-comment -DSASR -DWAV49 -DNDEBUG -I./inc'; gcc ${CFLAGS} ${SRC}; ar cr libgsm.a ${SRC/\.c/\.o}; ranlib libgsm.a; mkdir -p /usr/include/gsm /usr/lib; cp inc/*.h /usr/include/gsm; cp libgsm.a /usr/lib`

Reply:
Oh nice, also gcc supports multiple input files? didn't know. Yeah think i prefer something like that over base64, what do you think?

Is it possible to also include the default hardening flags https://github.com/wader/static-ffmpeg/blob/master/Dockerfile#L221 ?

Reply:
done as in the latest commit to the branch